### DISCUSSION TITLE
Use injectable from other library which populates with injectables from current project

### DISCUSSION THREAD

**Discussion by gdonoso94** [allrod5/injectable#109] [General]
Hi

I've been working on a command bus implenentation that executes a command using a command middleware chain.

This implementation is in one library separated from my project, just to be reusable, and it provides basic commands (for now a _dummy command_ that prints out the command object content) and basic command middlewares (for now just a _logging command middleware_ that prints out which command is going to be executed and a _handle command middleware_ that finds the appropriate command handler for the command and executes it.

The thing is that I would like to be able to add the commands defined in my project directly to the _handle command middleware_ and the middleware to be autowired to an injectable command bus from the library, and just use that injectable singleton in my project.

For now, the only way I managed to do this is, instead of making the command bus injectable, to create an instance of the command bus in my project and then use that, but then is not injectable and cannot be autowired... 

I'm working on a repo to demonstrate this and it will be ready soon. In the meantime... Any help is appreciated.

**Comment by roo-oliv** [comment]
Hi @gdonoso94! What is the problem when you try to make your command bus injectable? It should work provided you call [the `load_injection_container` function](https://injectable.readthedocs.io/en/latest/reference/injectable.html#injectable.load_injection_container) at the base `__init__.py` of your library and also at the startup of your project.

**Reply by gdonoso94** [reply]
Hi!

The problem is that the injection container seems to not contain the middlewares and handlers defined in my project. Is it being updated every once is called? Does it update every other injectable and autowired property on the project?

Thanks :)

**Reply by roo-oliv** [reply]
It is updated only when you call [load_injection_container](https://injectable.readthedocs.io/en/latest/reference/injectable.html#injectable.load_injection_container) with injectables found in all child paths where the file which called the method is located. So in case of injectables being declared both in the application and in the library, there is the need to call this method at the root of both.
